src/Symfony/Component/Debug/Tests/ExceptionHandlerTest.php

```diff
@@ -106,37 +106,78 @@ public function testNestedExceptions()
         $this->assertStringMatchesFormat('%A<p class="break-long-words trace-message">Foo</p>%A<p class="break-long-words trace-message">Bar</p>%A', $response);
     }
 
-    public function testHandle()
+    /**
+     * @dataProvider handleProvider
+     */
+    public function testHandle(\Throwable $exception, string $expectedClass, string $expectedTitle, string $expectedMessage)
     {
-        $exception = new \Exception('foo');
-
-        $handler = $this->getMockBuilder('Symfony\Component\Debug\ExceptionHandler')->setMethods(['sendPhpResponse'])->getMock();
-        $handler
-            ->expects($this->exactly(2))
-            ->method('sendPhpResponse');
+        $handler = new ExceptionHandler(true);
+        ob_start();
 
         $handler->handle($exception);
 
-        $handler->setHandler(function ($e) use ($exception) {
-            $this->assertSame($exception, $e);
+        $this->assertThatTheExceptionWasOutput(ob_get_clean(), $expectedClass, $expectedTitle, $expectedMessage);
+    }
+
+    public function handleProvider()
+    {
+        return [
+            [new \Exception('foo'), \Exception::class, 'Exception', 'foo'],
+            [new \Error('bar'), \Error::class, 'Error', 'bar'],
+        ];
+    }
+
+    public function testHandleWithACustomHandlerThatOutputsSomething()
+    {
+        $handler = new ExceptionHandler(true);
+        ob_start();
+        $handler->setHandler(function () {
+            echo 'ccc';
         });
 
-        $handler->handle($exception);
+        $handler->handle(new \Exception());
+        ob_end_flush(); // Necessary because of this PHP bug : https://bugs.php.net/bug.php?id=76563
```

Comment:
Lost more than 4 hours on this. If someone wants to look at it, just put this in a controller : 
```
$handler = ExceptionHandler::register();
$handler->setHandler(function ($e) {
    echo 'foo';
});

$handler->handle(new \RuntimeException('bar'));

dump(ob_get_clean()); die();
```

It will dump the full content (default error handler content + "foo"). If you comment the dump, only "foo" is actually rendered.